src/Npgsql.GeoJSON/GeoJSONHandler.cs

```diff
@@ -0,0 +1,792 @@
+
+        async ValueTask<Point> INpgsqlTypeHandler<Point>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (Point)await ReadGeometry(buf, async);
+
+        async ValueTask<LineString> INpgsqlTypeHandler<LineString>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (LineString)await ReadGeometry(buf, async);
+
+        async ValueTask<Polygon> INpgsqlTypeHandler<Polygon>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (Polygon)await ReadGeometry(buf, async);
+
+        async ValueTask<MultiPoint> INpgsqlTypeHandler<MultiPoint>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (MultiPoint)await ReadGeometry(buf, async);
+
+        async ValueTask<MultiLineString> INpgsqlTypeHandler<MultiLineString>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (MultiLineString)await ReadGeometry(buf, async);
+
+        async ValueTask<MultiPolygon> INpgsqlTypeHandler<MultiPolygon>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (MultiPolygon)await ReadGeometry(buf, async);
+
+        async ValueTask<GeometryCollection> INpgsqlTypeHandler<GeometryCollection>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (GeometryCollection)await ReadGeometry(buf, async);
+
+        async ValueTask<IGeoJSONObject> INpgsqlTypeHandler<IGeoJSONObject>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => await ReadGeometry(buf, async);
+
+        async ValueTask<IGeometryObject> INpgsqlTypeHandler<IGeometryObject>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription fieldDescription)
+            => (IGeometryObject)await ReadGeometry(buf, async);
+
+        async ValueTask<GeoJSONObject> ReadGeometry(NpgsqlReadBuffer buf, bool async)
+        {
+            var boundingBox = BoundingBox ? new BoundingBoxBuilder() : null;
+            var geometry = await ReadGeometryCore(buf, async, boundingBox);
+
+            geometry.BoundingBoxes = boundingBox?.Build();
+            return geometry;
+        }
+
+        async ValueTask<GeoJSONObject> ReadGeometryCore(NpgsqlReadBuffer buf, bool async, BoundingBoxBuilder boundingBox)
+        {
+            await buf.Ensure(SizeOfHeader, async);
+            var littleEndian = buf.ReadByte() > 0;
+            var type = (EwkbGeometryType)buf.ReadUInt32(littleEndian);
+
+            GeoJSONObject geometry;
+            NamedCRS crs = null;
+
+            if (HasSrid(type))
+            {
+                await buf.Ensure(4, async);
+                crs = GetCrs(buf.ReadInt32(littleEndian));
+            }
+
+            switch (type & EwkbGeometryType.BaseType)
+            {
+                case EwkbGeometryType.Point:
```

Comment:
nit: I usually indent case at the same level as the enclosing switch, i.e. like a goto label (which is what case labels really are). Similarly, I don't indent for curly braces *and* for the code (two levels of indentation), one is enough.

This reduces multiple levels of indentation

Reply:
Yeah, there's place where you can configure that (or at least with Resharper)